backend/src/test/java/codezap/template/service/ThumbnailServiceTest.java

```diff
@@ -0,0 +1,140 @@
+            var category = categoryRepository.save(CategoryFixture.getFirstCategory());
+            var template = templateRepository.save(TemplateFixture.get(member, category));
+            var sourceCode = sourceCodeRepository.save(new SourceCode(template, "Filename 1", "Content 1", 1));
+
+            sut.createThumbnail(template, sourceCode);
+            var actual = thumbnailRepository.findAll();
+
+            assertThat(actual).hasSize(1);
+        }
+    }
+
+    @Nested
+    @DisplayName("썸네일 조회")
+    class GetByTemplate {
+
+        @Test
+        @DisplayName("썸네일 조회 성공")
+        void getByTemplateSuccess() {
+            var member = memberRepository.save(MemberFixture.getFirstMember());
+            var category = categoryRepository.save(CategoryFixture.getFirstCategory());
+            var template = templateRepository.save(TemplateFixture.get(member, category));
+            var sourceCode = sourceCodeRepository.save(new SourceCode(template, "Filename 1", "Content 1", 1));
+            thumbnailRepository.save(new Thumbnail(template, sourceCode));
+
+            var actual = sut.getByTemplate(template);
+
+            assertThat(actual.getTemplate()).isEqualTo(template);
+        }
+
+        @Test
+        @DisplayName("썸네일 조회 실패: 해당하는 썸네일이 없는 경우")
+        void getByTemplateFailWithWrongID() {
+            var member = memberRepository.save(MemberFixture.getFirstMember());
+            var category = categoryRepository.save(CategoryFixture.getFirstCategory());
+            var template = templateRepository.save(TemplateFixture.get(member, category));
+
+            assertThatThrownBy(() -> sut.getByTemplate(template))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("식별자가 " + template.getId() + "인 템플릿에 해당하는 썸네일이 없습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("썸네일 삭제")
+    class DeleteByTemplateIDs {
+
+        @Test
+        @DisplayName("썸네일 삭제 성공")
+        void deleteByTemplateSuccess() {
+            var member = memberRepository.save(MemberFixture.getFirstMember());
+            var category = categoryRepository.save(CategoryFixture.getFirstCategory());
+            var template1 = templateRepository.save(TemplateFixture.get(member, category));
+            var sourceCode1 = sourceCodeRepository.save(new SourceCode(template1, "Filename 1", "Content 1", 1));
+            thumbnailRepository.save(new Thumbnail(template1, sourceCode1));
+            var template2 = templateRepository.save(TemplateFixture.get(member, category));
+            var sourceCode2 = sourceCodeRepository.save(new SourceCode(template2, "Filename 2", "Content 2", 1));
+            thumbnailRepository.save(new Thumbnail(template2, sourceCode2));
+
+            sut.deleteByTemplateIds(List.of(template1.getId()));
+            var actual = thumbnailRepository.findAll();
+
+            assertThat(actual).hasSize(1);
+            assertThat(actual.get(0).getTemplate()).isEqualTo(template2);
```

Comment:
```suggestion
            var member = memberRepository.save(MemberFixture.getFirstMember());
            var category = categoryRepository.save(CategoryFixture.getFirstCategory());
            var template1 = templateRepository.save(TemplateFixture.get(member, category));
            var sourceCode1 = sourceCodeRepository.save(new SourceCode(template1, "Filename 1", "Content 1", 1));
            var thumbnail1 = thumbnailRepository.save(new Thumbnail(template1, sourceCode1));
            var template2 = templateRepository.save(TemplateFixture.get(member, category));
            var sourceCode2 = sourceCodeRepository.save(new SourceCode(template2, "Filename 2", "Content 2", 1));
            var thumbnail2 = thumbnailRepository.save(new Thumbnail(template2, sourceCode2));

            sut.deleteByTemplateIds(List.of(template1.getId()));
            var actual = thumbnailRepository.findAll();

            assertThat(actual).hasSize(1)
                    .containsExactly(thumbnail2)
                    .doesNotContain(thumbnail1);
```

이부분도 체이닝으로 변경해 보았어요.